metrics/storage_collector.go

```diff
@@ -0,0 +1,124 @@
+package metrics
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-evm-gateway/models"
+)
+
+type StorageSizeCollector struct {
+	//TODO: think of adding error metric indicating we couldn't update the db size
+	storageDir  string
+	storageSize prometheus.Gauge
+	interval    time.Duration
+	logger      zerolog.Logger
+	ctx         context.Context
+	cancel      context.CancelFunc
+	ready       chan struct{}
+}
+
+type StorageSizeCollectorOpts struct {
+	Factory    promauto.Factory
+	Logger     zerolog.Logger
+	StorageDir string
+	Interval   time.Duration
```

Comment:
Don't expose these
```suggestion
	Factory    promauto.Factory
	Logger     zerolog.Logger
	StorageDir string
	Interval   time.Duration
```

Reply:
I'm not sure why this changed now, couldn't you just pass in the collector like for others?